### PR TITLE
Convert TRDGeometry to a singleton O2-1197

### PR DESCRIPTION
**Pull request by bazinski**
Convert TRDGeometry to a singleton.
1. removing the TRDGeometry instance in every Tracklet.
2. fix where TRDGeometry is used in various places.
3. move Tracklet functions into TRDGeometry, moving the geometry logic out of tracklets and where it should live.
4. Add some tests for the added functions.

### CODE REVIEW

**Comment by sawenzel** [comment]
You need to fix some macros as well:
```
In file included from input_line_12:6:
/mnt/mesos/sandbox/sandbox/sw/SOURCES/O2/3011/0/GPU/GPUTracking/Standalone/tools/createGeo.C:15:24: error: calling a private constructor of class 'o2::trd::TRDGeometry'
  o2::trd::TRDGeometry gm;
                       ^
/mnt/mesos/sandbox/sandbox/sw/SOURCES/O2/3011/0/Detectors/TRD/base/include/TRDBase/TRDGeometry.h:66:3: note: declared private here
  TRDGeometry();
  ^
Error in <ACLiC>: Dictionary generation failed!
```

**Comment by bazinski** [comment]
@sawenzel oui, that never came up when testing on my desktop, will fix.
was not sure whether to put TRDGeometry() back or fix the script.

File: Detectors/TRD/workflow/CMakeLists.txt [hunk 1]
```diff
@@ -8,6 +8,7 @@
 # granted to it by virtue of its status as an Intergovernmental Organization or
 # submit itself to any jurisdiction.
 
+add_compile_options(-O0 -g -fPIC)
```
Comment (sawenzel) [inline Detectors/TRD/workflow/CMakeLists.txt]: Please don't commit this.

File: Detectors/TRD/workflow/src/TRDDigitizerSpec.cxx [hunk 2]
```diff
@@ -72,10 +72,10 @@ class TRDDPLDigitizerTask
       mSimChains.emplace_back(new TChain("o2sim"));
       mSimChains.back()->AddFile(signalfilename.c_str());
     }
-
     if (!gGeoManager) {
       o2::base::GeometryManager::loadGeometry();
     }
+    LOG(INFO) << "initialed TRD digitization";
```
Comment (sawenzel) [inline Detectors/TRD/workflow/src/TRDDigitizerSpec.cxx]: typo

Comment (bazinski) [reply Detectors/TRD/workflow/src/TRDDigitizerSpec.cxx]: I only speak the language, this writing stuff is hard ....

File: Detectors/TRD/workflow/CMakeLists.txt (hunk 1 above)
Comment (bazinski) [reply Detectors/TRD/workflow/CMakeLists.txt]: oops ....

**Comment by bazinski** [comment]
will squash the commits and then someone can approve and merge

**Comment by bazinski** [comment]
Now with a simpler implementation of singleton pattern. Thanks @sawenzel

**Pull request merged**